discord/discordDeleteMessage.js

```diff
@@ -0,0 +1,96 @@
+module.exports = function (RED) {
+  var discordBotManager = require('./lib/discordBotManager.js');
+
+  function discordDeleteMessage(config) {
+    RED.nodes.createNode(this, config);
+    var node = this;
+    var configNode = RED.nodes.getNode(config.token);
+    discordBotManager.getBot(configNode).then(function (bot) {
+      node.on('input', function (msg) {
+        if (!msg.id || !msg.channel) {
+          node.error(`Either msg.id or msg.channel wasn't set`);
+          node.status({
+            fill: "red",
+            shape: "dot",
+            text: `msg.id and msg.channel are required`
+          })
+          return;
+        }
+
+        var msgId = msg.id;
+        if (msgId && typeof msgId !== 'string') {
+          if (msgId.hasOwnProperty('id')) {
+            msgId = msgID.id;
```

Comment:
This will throw a referenceError as `msgID` is not defined.
`msgID.id` -> `msgId.id`